server/src/lib/model/leaderboard.ts

```diff
@@ -15,6 +15,47 @@ const bucket = new Bucket(model, s3);
 
 const db = getMySQLInstance();
 
+const team_only_condition = `
+(visible OR (visible = 0 AND user_clients.client_id = ?))
+AND enroll.team_id = (
+  SELECT team_id FROM enroll e LEFT JOIN challenges c ON e.challenge_id = c.id WHERE e.client_id = ? AND c.url_token = ?
+)`;
+const general_condition = `
+(
+  (visible = 0 AND user_clients.client_id = ?)
+  OR (visible = 1)
+  OR (visible = 2 AND teams.id = (
+      SELECT team_id FROM enroll e LEFT JOIN challenges c ON e.challenge_id = c.id WHERE e.client_id = ? AND c.url_token = ?
+  ))
+)`;
```

Comment:
Not a blocker by any means, but it might be nice to combine some of this:
```
const self_subcondition = '(visible = 0 AND user_clients.client_id = ?)';
const teammate_subcondition =
	'(SELECT team_id FROM enroll e LEFT JOIN challenges c ON e.challenge_id = c.id WHERE e.client_id = ? AND c.url_token = ?)'
const team_only_condition =
	`(visible OR ${self_subcondition}) AND enroll.team_id = ${teammate_subcondition}`;
const general_condition = `(
  ${self_subcondition}
  OR (visible = 1)
  OR (visible = 2 AND teams.id = ${teammate_subcondition})
)`;
```

Reply:
Great, it is cleaner.